core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedService.java

```diff
@@ -141,6 +143,16 @@
             responseType = ResponseType.OTHER_OBJECTS;
         }
 
+        ServiceName serviceName = method.getAnnotation(ServiceName.class);
+        if (serviceName == null) {
+            serviceName = object.getClass().getAnnotation(ServiceName.class);
+        }
```

Comment:
Better using `AnnotationUtil`: https://github.com/line/armeria/blob/master/core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotationUtil.java

Reply:
Thanks for the nice tip. 😀